### DISCUSSION TITLE
colmap2nerf.py gets killed by oom killer

### DISCUSSION THREAD

**Discussion by ivansabolic** [NVlabs/instant-ngp#956] [Q&A]
Cheers,

As the title says, I am having problem running colmap2nerf.py script on my input video. My input video is in resolution 3840 x 2160 and I am running the script on a setup with 32GB RAM and 2 Nvidia 3080 ti GPUs. I though that amount of RAM would be enough for an image of this size, but unfortunately that is not the case. If I resize the image to 1920 x 1080, the script is finished successfully.

Any ideas on how to approach this? Should I be looking for a setup with more RAM? Or should I give up on that image size and focus on smaller images?

Thanks!

**Comment by toomy0toons** [comment]
hi i know this question has not been answered for a long time but i  leave comment for future users

approches you could try instead of resizing image

1. changing the number of images to match (reduce total mem usage)
-> pass in fewer number for --fps option in colmap2nerf.py so that you extract less image

colmap uses most mem at feature matching stage so you could try passing colmap args in colmap2nerf.py
```
# lines 123, 124 in colmap2nerf.py
# append options here
	do_system(f"{colmap_binary} feature_extractor --ImageReader.camera_model {args.colmap_camera_model} --ImageReader.camera_params \"{args.colmap_camera_params}\" --SiftExtraction.estimate_affine_shape=true --SiftExtraction.domain_size_pooling=true --ImageReader.single_camera 1 --database_path {db} --image_path {images}")
	match_cmd = f"{colmap_binary} {args.colmap_matcher}_matcher --SiftMatching.guided_matching=true --database_path {db}"
```

2. use GPU instead of CPU in feature matching stage
-> pass --SiftExtraction.gpu_index=0,1 to line 123(extraction) and --SiftMatching.gpu_index=0,1 to 124(matching) args in colmap2nerf.py 

3.  Try decreasing the option --SiftMatching.max_num_matches in colmap
-> same as above, pass --SiftMatching.max_num_matches 1000~10000 (any number that fits your mem) in colmap2nerf.py line 124(matching)

checkout https://colmap.github.io/faq.html for more available options

hope it helps

**Reply by schnittbrot** [reply]
For this to work you have to remove the following arguments from line 123 in colmap2nerf.py as they cause a CPU fallback:

--SiftExtraction.estimate_affine_shape=true --SiftExtraction.domain_size_pooling=true
